clients/client-python/gravitino/auth/auth_data_provider.py

```diff
@@ -0,0 +1,31 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
+
+from abc import ABC, abstractmethod
+
+
+class AuthDataProvider(ABC):
```

Comment:
It seems that this class has extended `Closeable` interface in Java and `close()` method would be used in other kinds of auth provider, so I think it is better to also support it here.